sale_order_general_discount_payment_term/views/res_partner.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" ?>
+<odoo>
+    <record id="view_partner_form_hide_discount" model="ir.ui.view">
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="sale_order_general_discount.view_partner_form" />
+        <field name="arch" type="xml">
+            <xpath
+                expr="//page[@name='sales_purchases']//label[@for='sale_discount']"
+                position="attributes"
+            >
+                <attribute name="invisible">1</attribute>
```

Comment:
This is not correct. This is out of scope of the module. And there's already a group for hiding this for those you don't want to see it.

Reply:
@pedrobaeza the point of this change is that if there is a general discount set in partner and another general discount set in payment term they will be in conflict for setting the general discount on SO. 

How else do you suggest we avoid this scenario?

Reply:
Sorry, the flow is unclear to me. What happens if partner 1 has a default general discount set and in SO user selects a payment term with a general discount? Which general discount should be applied?

With this module installed, it's very likely that the only general discount that should be applied is the payment term one. 

Shall we make it easier with a flag in settings allowing user to decide whether to hide the general discount in partner field?

Reply:
You should clear that general discount on payment term assignment IMO.

Reply:
The question is that you shouldn't remove the possibility of assigning directly a general discount for all cases. Just for those covered by this module (with payment term). If not, you will force to assign a payment term always if you want a general discount.

Reply:
User has the possibility of assigning general discount in SO in 3 ways: 

1) default from partner
2) manually from field "discount" in SO
3) default from payment term

base module "sale_order_general_discount" implements the first two options; this module implements the third.

I think if we add a "hide default general discount in partner" flag in settings to this module we will have the cleaner implementation as the user can decide:

- whether to keep all 3 options,
- to keep only 2 and 3 (flag enabled), or 
- to keep 1 and 2 by not installing this module. 

Could that work?

Reply:
I still think that you don't need another extra config. Just don't hide the field if you don't have a payment term selected.

Reply:
Alright, @odooNextev can you implement that?

Reply:
@pedrobaeza with the latest change by @odooNextev , if payment term discount = 0 and partner default discount is set, partner default is used in SO; otherwise payment term discount is used. Could this work?